tests/integration/test_charm.py

```diff
@@ -64,12 +86,62 @@ async def test_build_and_deploy(
         },
     )
     await action.wait()
-    await ops_test.model.add_relation(penpot.name, postgresql_k8s.name)
+    await ops_test.model.add_relation("self-signed-certificates", "nginx-ingress-integrator")
+    await ops_test.model.add_relation(penpot.name, "postgresql-k8s")
     await ops_test.model.add_relation(penpot.name, redis_k8s.name)
     await ops_test.model.add_relation(penpot.name, s3_integrator.name)
     await ops_test.model.add_relation(penpot.name, f"{smtp_integrator.name}:smtp")
     await ops_test.model.add_relation(penpot.name, nginx_ingress_integrator.name)
-    await ops_test.model.wait_for_idle(timeout=900, status="active")
+    await ops_test.model.wait_for_idle(timeout=900, status="active", raise_on_error=False)
+    logger.info(
+        "test user account: (%s, %s)", ext_idp_service.user_email, ext_idp_service.user_password
+    )
+    action = (
+        await ops_test.model.applications["self-signed-certificates"]
+        .units[0]
+        .run_action("get-ca-certificate")
+    )
+    await action.wait()
+    ca_cert: str = action.results["ca-certificate"]
+    for unit in range(num_units):
+        logger.info("copying oauth ca cert into penpot/%s", unit)
+        await ops_test.juju(
```

Comment:
Can't we use `ops_test.run` ?

Reply:
`ops_test.juju(...)` ≈ `ops_test.run("juju", ...)`